lib/connectors/AsyncSentinelConnector.ts

```diff
@@ -0,0 +1,35 @@
+import SentinelConnector, {
+    ISentinelConnectionOptions,
+    ISentinelConnectorOptions,
+    ISentinelAddress,
+    SentinelIterator,
+    EMPTY_SENTINELS_MSG } from './SentinelConnector';
+import { ErrorEmitter } from './AbstractConnector';
+import { NetStream } from '../types';
+import { NodeCallback } from './types';
+
+export type AsyncSentinelFetch = () => Promise<Partial<ISentinelAddress>[]>
+
+export default class AsyncSentinelConnector extends SentinelConnector {
+    private fetch: AsyncSentinelFetch
+
+    constructor(options: ISentinelConnectionOptions, fetch: AsyncSentinelFetch) {
+        options.sentinels = options.sentinels || [{host: 'localhost', port: 6379}] // Placeholder
```

Comment:
What about not calling `super()` here (so that we don't need a placeholder)?

Reply:
you always have to call super though